wolfcrypt/src/asn.c

```diff
@@ -12248,7 +12248,11 @@ int GetAsnTimeString(void* currTime, byte* buf, word32 len)
         hour = ts->tm_hour;
         mini = ts->tm_min;
         sec  = ts->tm_sec;
-        XSNPRINTF((char *)utc_str, ASN_UTC_TIME_SIZE,
+        #if defined(WOLF_C89)
+            XSPRINTF((char *)utc_str,
```

Comment:
I think this is sufficiently safe the way you've refactored it -- the output string has a known length based on the format, and you're using a properly descriptive macro.